packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -0,0 +1,505 @@
+            { failure: !!inputError },
+            'input'
+          )}
+          value={inputValue}
+          onInput={(e) => onInputValueChange(e.target?.value?.trim())}
+        />
+      }{' '}
+      # of all requirements
+    </span>
+  );
+
+  return (
+    <CWRadioButton
+      label={Label}
+      value={REQUIREMENTS_TO_FULFILL.N_REQUIREMENTS}
+      name="requirementsToFulfill"
+      hookToForm
+    />
+  );
+};
+
+const MAX_REQUIREMENTS = 10;
+
+const getRequirementSubFormSchema = (
+  requirementType: string
+): ZodObject<any> => {
+  const isTokenRequirement = Object.values(TOKENS).includes(requirementType);
+  const schema = isTokenRequirement
+    ? requirementSubFormValidationSchema.omit({
+        requirementContractAddress: true,
+      })
+    : requirementSubFormValidationSchema;
+  return schema;
+};
+
+const GroupForm = ({
+  formType,
+  onSubmit,
+  initialValues = {},
+}: GroupFormProps) => {
+  const { data: topics } = useFetchTopicsQuery({
+    chainId: app.activeChainId(),
+  });
+  const sortedTopics = (topics || []).sort((a, b) => a?.name?.localeCompare(b));
+  const [cwRequiremenetsLabelInputField, setCwRequiremenetsLabelInputField] =
+    useState<CWRequirementsLabelInputFieldState>({ value: '1', error: '' });
+  const [requirementSubForms, setRequirementSubForms] = useState<
+    RequirementSubFormsState[]
+  >([
+    {
+      values: {
+        requirementAmount: '',
+        requirementChain: '',
+        requirementCondition: '',
+        requirementContractAddress: '',
+        requirementType: '',
+      },
+      errors: {},
+    },
+  ]);
+
+  useEffect(() => {
+    if (initialValues.requirements) {
+      setRequirementSubForms(
+        initialValues.requirements.map((x) => ({
+          defaultValues: x,
+          values: {
+            requirementAmount: x?.requirementAmount || '',
+            requirementChain: x?.requirementChain?.value || '',
+            requirementCondition: x?.requirementCondition?.value || '',
+            requirementContractAddress: x?.requirementContractAddress || '',
+            requirementType: x?.requirementType?.value || '',
+          },
+          errors: {},
+        }))
+      );
+    }
+
+    if (
+      initialValues.requirementsToFulfill &&
+      initialValues.requirementsToFulfill !==
+        REQUIREMENTS_TO_FULFILL.ALL_REQUIREMENTS
+    ) {
+      setCwRequiremenetsLabelInputField({
+        ...cwRequiremenetsLabelInputField,
+        value: `${initialValues.requirementsToFulfill}`,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const removeRequirementByIndex = (index: number) => {
+    setRequirementSubForms(requirementSubForms.splice(index, 1));
+  };
+
+  const [requirementLimit, setRequirementLimit] = useState(false);
+
+  const addRequirementSubForm = () => {
+    if (requirementSubForms.length >= MAX_REQUIREMENTS) {
+      setRequirementLimit(true);
+    }
```

Comment:
https://github.com/hicommonwealth/commonwealth/assets/51641047/01b120bc-a95b-4438-b1bf-72eaccc76571

I am still able to reproduce [this](https://github.com/hicommonwealth/commonwealth/pull/5434#issuecomment-1781457958) issue.

Also if we add 10 requirements and then remove some, the button remains disabled.